### PR TITLE
Avoid HTML-encoding HTML on contextContent attributes

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
We're HTML-encoding the "forgot password" link when a user is changing their own email address. We've got a couple of other places that are passing HTML as a string to contextContent.

#### Changes
* Use HtmlString to avoid showing raw HTML to user

### CODE REVIEW

File: core/src/org/labkey/core/view/styleGuide.jsp [hunk 1]
```diff
@@ -276,7 +278,8 @@
         </div>
         <br>
         <h2>Context Content</h2>
-        <% String helpText = "Go to home: <a href=" + ActionURL.getBaseServerURL() + ">Server Home</a>";%>
+        <% HtmlString helpText = DOM.createHtml(DOM.createHtmlFragment("Go to home: ", DOM.A(DOM.at(DOM.Attribute.href, ActionURL.getBaseServerURL()), "Server Home"))); %>
```
Comment (labkey-adam) [inline core/src/org/labkey/core/view/styleGuide.jsp:281]: Could use `link()` here

File: core/src/org/labkey/core/user/changeEmail.jsp [hunk 2]
```diff
@@ -60,7 +63,7 @@
     }
     else if (form.getIsPasswordPrompt())
     {
-        String resetPasswordLink = "<a href=" + h(urlFor(ResetPasswordAction.class)) + ">forgot password</a>";
+        HtmlString resetPasswordLink = DOM.createHtml(DOM.A(DOM.at(DOM.Attribute.href, urlFor(ResetPasswordAction.class)), "forgot password"));
```
Comment (labkey-adam) [inline core/src/org/labkey/core/user/changeEmail.jsp]: Could use `link()` here

**Review by labkey-adam — APPROVED** [review]
Optional suggestions

File: core/src/org/labkey/core/user/changeEmail.jsp (hunk 2 above)
Comment (labkey-jeckels) [reply core/src/org/labkey/core/user/changeEmail.jsp]: Converted

File: core/src/org/labkey/core/view/styleGuide.jsp (hunk 1 above)
Comment (labkey-jeckels) [reply core/src/org/labkey/core/view/styleGuide.jsp:281]: I fought with this a bit and gave up. link() needs a URLHelper, not a String URL, and I had to append the "Go to home:" prefix, so it didn't end up being any leaner. FWIW, I find it incredibly awkward to find the right way to nest Renderable vs Appendable vs builders vs HtmlString

**Pull request merged**